### DISCUSSION TITLE
I've created a python repl client library.

### DISCUSSION THREAD

**Discussion by spierepf** [micropython/micropython#11798] [Show and tell]
I originally mentioned the idea here:
https://forum.micropython.org/viewtopic.php?t=12910&p=70298

And finally I've created a very first cut:

https://github.com/spierepf/repl_client

If you're interested, please take a look and let me know what you think.

**Comment by peterhinch** [comment]
See the official [mpremote](http://docs.micropython.org/en/latest/reference/mpremote.html) which works well and is under active development.

**Reply by sethstenzel** [reply]
I was looking at this recently. Is there a way to use mpremote as a library instead of a cli utility? The documentation makes me think no at the moment. If I open a repl and import, I can import it, but it has no available methods so maybe there is just no documentation for importing and using it as a library.

**Reply by jimmo** [reply]
See https://github.com/micropython/micropython/pull/11777 (and https://github.com/micropython/micropython/pull/11669 before it) -- making `mpremote` more useful as a library is something that's being actively worked on at the moment.

**Reply by gitcnd** [reply]
This has it inbuilt: https://github.com/vsolina/micropython-web-editor (repl over http wifi)

**Comment by Wind-stormger** [comment]
or [mpbridge](https://github.com/AmirHmZz/mpbridge)
